src/fluka/fluka_funcs.cpp

```diff
@@ -1027,17 +1037,18 @@ void fludag_all_tallies(std::ostringstream& mstr, std::map<std::string, pyne::Ta
 // fludag_all_materials
 //---------------------------------------------------------------------------//
 // Get material cards for all materials in the problem, both elemental and compounds
-void fludag_all_materials(std::ostringstream& mstr, std::map<std::string, pyne::Material> pyne_map) {
+void fludag_all_materials(std::ostringstream& mstr,
+                          const pyne::mat_map& pyne_map) {
   std::set<int> exception_set = make_exception_set();
 
   std::map<int, std::string> map_nucid_fname;
 
   pyne::Material unique = pyne::Material();
 
   // loop over all materials, summing
-  std::map<std::string, pyne::Material>::iterator nuc;
+  pyne::mat_map::const_iterator nuc;
   for (nuc = pyne_map.begin(); nuc != pyne_map.end(); ++nuc) {
-    unique = unique + (nuc->second);
+    unique = unique + *(nuc->second);
   }
   // now collapse elements
   unique = unique.collapse_elements(exception_set);
```

Comment:
I am not 100% on what is done there.
but with the Pyne::MaterialLibrary you can use `get_nuclist()` to get a set of the nuclides (if required...)

Reply:
well thats all that does, but that functionality from materialibrary must be new?

Reply:
Yeah I'm honestly not sure about what's being done here either. In terms of the objects, it's iterating over the material objects and adding them. `nuc` is an odd variable name though. You ok with a change to `it` as well here?

Reply:
what it does is use pyne::material addition to build a material object that has every nuclide present in the entire problem.

Reply:
that 'special' unique material then does collapse_elements() to get a material that contains the special set of nuclides that FLUKA is aware of

Reply:
but that material is never written

Reply:
nuc should be material or mat or something like that

Reply:
> makeclean 
> well thats all that does, but that functionality from materialibrary must be new?

Yes it has been added when we created the C++ API

Reply:
Done!